src/structures/GuildTemplate.js

```diff
@@ -0,0 +1,197 @@
+'use strict';
+
+const Base = require('./Base');
+const { Events } = require('../util/Constants');
+const DataResolver = require('../util/DataResolver');
+
+/**
+ * Represents the template for a guild.
+ * @extends {Base}
+ */
+class GuildTemplate extends Base {
+  /**
+   * @param {Client} client The instantiating client
+   * @param {Object} data The raw data for the template
+   */
+  constructor(client, data) {
+    super(client);
+    this._patch(data);
+  }
+
+  /**
+   * Builds or updates the template with the provided data.
+   * @param {Object} data The raw data for the template
+   * @returns {GuildTemplate}
+   * @private
+   */
+  _patch(data) {
+    /**
+     * The unique code of this template
+     * @type {string}
+     */
+    this.code = data.code;
+
+    /**
+     * The name of this template
+     * @type {string}
+     */
+    this.name = data.name;
+
+    /**
+     * The description of this template
+     * @type {?string}
+     */
+    this.description = data.description;
+
+    /**
+     * The amount of times this template has been used
+     * @type {number}
+     */
+    this.usageCount = data.usage_count;
+
+    /**
+     * The ID of the user that created this template
+     * @type {string}
+     */
+    this.creatorID = data.creator_id;
+
+    /**
+     * The user that created this template
+     * @type {User}
+     */
+    this.creator = this.client.users.add(data.creator);
+
+    /**
+     * The time of when this template was created at
+     * @type {Date}
+     */
+    this.createdAt = new Date(data.created_at);
+
+    /**
+     * The time of when this template was last synced to the guild
+     * @type {Date}
+     */
+    this.updatedAt = new Date(data.updated_at);
+
+    /**
+     * The ID of the guild that this template belongs to
+     * @type {string}
+     */
+    this.guildID = data.source_guild_id;
+
+    /**
+     * The guild that this template belongs to
+     * @type {Guild}
+     */
+    this.guild = this.client.guilds.add(data.serialized_source_guild, false);
```

Comment:
This will insert the template channels into `Client#channels`, ignoring the cache parameter.

https://github.com/discordjs/discord.js/blob/eaecd0e8b790f218b2c52021b433c0957ec539c3/src/structures/Guild.js#L396-L401

Not too sure how to go about this, we could go with just the raw object. Probably not the most handy thing to work with however, not that I can think of many use cases.

Reply:
I think it's a bit misleading that `guildID` and `guild` represent two different guilds, it probably is better to rename this field to something that properly reflects that this is the actual template, not "just" the source guild itself.

Reply:
Renamed the prop to `serializedGuild`, also went with just returning the raw object and threw in a link on the JSDoc to the guild structure over on the API docs.